modules/renter/persist_test.go

```diff
@@ -317,3 +317,28 @@ func TestSiafileCompatibility(t *testing.T) {
 		t.Fatal("nickname not loaded properly:", names)
 	}
 }
+
+// TestUpgradeLegacyPersistFile
+func TestUpgradeLegacyPersistFile(t *testing.T) {
+	// Create renter
+	r := &Renter{}
+	r.persistDir = filepath.Join("..", "..", "persist", "testdata")
+
+	// Set persist version to legacy version and save persist file
+	settingsMetadata.Version = persistVersion040
+	data := struct {
+		Tracking  map[string]string
+		Repairing map[string]string
+	}{}
+	persist.SaveJSON(settingsMetadata, data, filepath.Join(r.persistDir, PersistFilename))
+
+	// Confirm loading of legacy persist file
+	r.persist = persistence{
+		Tracking: make(map[string]trackedFile),
+	}
+	settingsMetadata.Version = persistVersion
+	err := r.load()
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
After loading, need to check that the new values were properly set to the default values for those settings.